### DISCUSSION TITLE
询问最新版本的主干代码编译工具链版本

### DISCUSSION THREAD

**Discussion by YangQi1234** [QQxiaoming/quard_star_tutorial#23] [Q&A]
询问最新版本编译工具链是哪一个版本呢？我目前在ubuntu20.04上编译用最新的toolchain编译mask_rom出错，最后想恳请下能不能上传一个编译环境虚拟机会比较方便后续学习和交流扩展，因为我在编译过程中确实遇到很多难搞的编译问题，万分感谢。
<img width="431" alt="1687097063769" src="https://github.com/QQxiaoming/quard_star_tutorial/assets/36604073/c7dce1a3-576a-43d9-9e7a-333bca2c0d5d">

**Comment by QQxiaoming** [comment]
感谢提问。

目前我在github action上用的编译工具链是可以保证主线代码都能通过编译的，你可以参考 https://github.com/QQxiaoming/quard_star_tutorial/issues/21#issuecomment-1455831281 ，直接下载可以使用的二进制工具链。

另外我自己使用的工具链是自己编译的gnu主线上的，可以参考[文档](https://quard-star-tutorial.readthedocs.io/zh_CN/latest/ch22.html)自行编译。
